src/helpers/dynamic-column-headers.ts

```diff
@@ -0,0 +1,94 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { JPath } from './jpath';
+
+export class DynamicColumnHeaders {
+  headers: Array<string>;
+  static namespaceSeparator = ':';
+
+  /**
+   * Constructor.
+   *
+   * @param {Array<string>} headers
+   */
+  constructor(headers: Array<string>) {
+    this.headers = headers;
+  }
+
+  /**
+   * Maps values of cells in a row to their respective
+   * header for a given namespace.
+   * With no namespace provided, it returns values for ALL namespaces
+   * and nests the results under each namespace as property.
+   * Per default it will nest results under the respective group, which is '0'
+   * if none exists.
+   *
+   * @param {Array<string>} row
+   * @param {string} namespace
+   * @param {string} separator
+   * @param {boolean} includeGroup
+   * @returns {Record<string, Object>}
+   */
+  getMappedValues(
+    row: Array<string>,
+    namespace: string | undefined = undefined,
+    includeGroup: boolean = true
+  ): Record<string, any> {
```

Comment:
Is the value type of record really `any` or can it be defined as a subset of any? E.g.:
```
type CellValue = string | number | boolean | ...
```

Reply:
Specifying the type here would break `index.ts` L124-L131 (`source.url`, among others)